vyos_modular/commands.py

```diff
@@ -91,3 +102,32 @@ def apply_patch(patch_path: pathlib.Path, dst: pathlib.Path):
     ret = _run_command(["git", "apply", str(patch_path.resolve())], cwd=dst)
     if ret != 0:
         raise RuntimeError("Failed to apply patch")
+
+
+def download_package(packages_dir: pathlib.Path, package_url: PackageURL):
+    package_filename = package_url.filename
+    if package_filename is None:
+        package_filename = package_url.url.split("/")[-1]
+        if not package_filename.endswith("deb"):
+            raise RuntimeError("Cannot determine package filename after download")
+
+    package_path = packages_dir / package_filename
+
+    print(f"INFO: Downloading {package_url.url} to {package_path}")
+    urllib.request.urlretrieve(package_url.url, package_path)
+
+
+# Execute the contents of `script_body` as a script using bash with build_dir as the CWD
+def run_script(build_dir: pathlib.Path, current_build_name: str, script_body: str):
```

Comment:
As this could be seen as a potentially dangerous action (running arbitrary bash code), I'd like to see the default action to prompt the user to allow the script to run, and introduce either a config option or a CLI flag to automatically accept